client/src/components/Room.jsx

```diff
@@ -78,12 +89,13 @@ class Room extends React.Component {
     } else {
       const instMap=this.state.mapping;
       const keyPressed= e.key.toUpperCase();
-      const note=instMap[keyPressed][1];
-      const octave=instMap[keyPressed][2];
-      const pd=instMap[keyPressed][3];
-      const type=instMap[keyPressed][4];
+      const sequence=JSON.parse(instMap[keyPressed]);
+      const note=sequence[1];
```

Comment:
You can use array destructuring to assign variables in ES6: [See this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)